mod-director-core/src/main/java/net/jan/moddirector/core/configuration/type/CurseRemoteMod.java

```diff
@@ -32,11 +36,15 @@ public CurseRemoteMod(
             @JsonProperty(value = "addonId", required = true) int addonId,
             @JsonProperty(value = "fileId", required = true) int fileId,
             @JsonProperty(value = "metadata") RemoteModMetadata metadata,
-            @JsonProperty(value = "options") Map<String, Object> options
+            @JsonProperty(value = "options") Map<String, Object> options,
+            @JsonProperty(value = "folder") String folderName,
+            @JsonProperty(value = "inject") String inject
```

Comment:
Use `boolean` instead of `String`, Jackson will doe the parsing for you.

However, in this case you might want to use `java.lang.Boolean`, so that the value can be `null`.

Use the following logic in the constructor of `ModDirectorRemoteMod` to then detect the `inject` option value:
```java
if(inject == null) {
    this.inject = folderName == null;
} else {
    this.inject = inject;
}
```